mlir_graphblas/src/lib/GraphBLAS/GraphBLASUtils.cpp

```diff
@@ -538,3 +541,104 @@ LogicalResult ExtensionBlocks::extractBlocks(Operation *op, RegionRange &regions
 
   return success();
 };
+
+LogicalResult populateSemiringRegions(OpBuilder &builder, Location loc,
+                                      StringRef semiring, Type valueType,
+                                      RegionRange regions)
+{
+  // Insert additive identity
+  Region *addIdentityRegion = regions[0];
+  Value addIdentity;
+  /*Block *addIdentityBlock = */ builder.createBlock(addIdentityRegion, {}, {});
+  if (semiring == "plus_pair" || semiring == "plus_times" || semiring == "plus_plus")
+  {
+    // Add identity
+    addIdentity = llvm::TypeSwitch<Type, Value>(valueType)
+                      .Case<IntegerType>([&](IntegerType type)
+                                         { return builder.create<ConstantIntOp>(loc, 0, type.getWidth()); })
+                      .Case<FloatType>([&](FloatType type)
+                                       { return builder.create<ConstantFloatOp>(loc, APFloat(0.0), type); });
+  }
+  else if (semiring == "min_plus")
+  {
+    addIdentity = llvm::TypeSwitch<Type, Value>(valueType)
+                      .Case<IntegerType>([&](IntegerType type)
+                                         { return builder.create<ConstantOp>(loc, builder.getIntegerAttr(valueType, APInt::getSignedMaxValue(type.getWidth()))); })
+                      .Case<FloatType>([&](FloatType type)
+                                       { return builder.create<ConstantOp>(loc, builder.getFloatAttr(valueType, APFloat::getLargest(type.getFloatSemantics(), false))); });
+  }
+  else
```

Comment:
Usually, for a single statement, MLIR does not use {...}. I would also consider using the same style that MLIR uses using clang-format (LLVM style: https://github.com/llvm/llvm-project/blob/main/mlir/.clang-format)